### DISCUSSION TITLE
How to use ebusctl decode

### DISCUSSION THREAD

**Discussion by gperdach** [john30/ebusd#662] [Q&A]
Hello,
I'd like to test decoding some messages from the ebus, but I don't know how to use it.
With ebusctl decode I tried this:
**ebusctl decode "r,,DateTime,Datum Uhrzeit,,,B504,00,,,dcfstate;btime;bdate;temp2,,," 1008b50401000a0302470601100622100d**
and some similar commands. For example without quotes and without the message part.
Can you help me, how I can use the decode feature? 
Regards,
Gregor

**Comment by john30** [comment]
as stated in the [help of the decode command](https://github.com/john30/ebusd/wiki/3.1.-TCP-client-commands#decode), you can only pass the section beginning with the first field type and pass in the appropriate data part, i.e. in your example:
`ebusctl decode "dcfstate;btime;bdate;temp2" 0302470601100622100d`

**Comment by gperdach** [comment]
Thanks, this was working.
Is there a way to decode like 
`ebusctl grab result decode`
for a single value to get all possible combinations?
For example: 
`ebusctl decodevalue 0003`
to get
` BCD   00=0, 03=3
 BCD:2 0003=300
 BCD:3 0003=300
 BCD:4 0003=300
 BDA   0003="08.02.1902"
 BDA:3 0003="08.02.1902"
 BDY   00=Mon, 03=Thu
 BDZ   0003="08.02.1902"
 BTI   0003="03:00"
 BTM   0003="03:00"
 D1B   00=0, 03=3
 D1C   00=0.0, 03=1.5
 D2B   0003=3.000
 D2C   0003=48.00
 DAY   0003="08.02.1902"
 EXP   0003=1.0762e-42
 EXR   0003=4.2039e-45
 FLR   0003=0.003
 FLT   0003=0.768
 HCD   0003=300
 HCD:1 00=0, 03=3
 HCD:2 0003=300
 HCD:3 0003=300
 HDA   0003="08.02.1902"
 HDA:3 0003="08.02.1902"
 HDY   00=Tue, 03=Fri
 HEX:2 0003="00 03"
 HTI   0003="00:03"
 HTM   0003="00:03"
 MIN   0003="12:48"
 NTS:2 0003=""
 PIN   0003=0003
 S3N   0003=768
 S3R   0003=3
 SCH   00=0, 03=3
 SIN   0003=768
 SIR   0003=3
 SLG   0003=768
 SLR   0003=3
 STR:2 0003=""
 TEM_P 0003=06-000
 TTM   00="00:00", 03="00:30"
 U3N   0003=768
 U3R   0003=3`

**Comment by cyberthom42** [comment]
I have the same question. Lets assume I see the following message in the log:
`received unknown MS cmd: 7108b51a0304d732 / 0ed707c0cfc07b3c00b01b20000000`
Is there a way to perform a decode similar to the grab result decode, something like this (but this does not work):
`ebusctl decode 0ed707c0cfc07b3c00b01b20000000`
